### PR TITLE
EMIRT push

### PR DESCRIPTION
**Pull request by SiHhangChen**
Thanks for sending a pull request! 
Please make sure you click the link above to view the [contribution guidelines](../blob/master/CONTRIBUTE.md), 
then fill out the blanks below.

## Description ##
The EM algorithm part of the IRT model has been completed and is now submitted

### What does this implement/fix? Explain your changes.
Completed part of the code on the EM algorithm in the IRT model, and provided several functions such as fit, predict, predict_proba, save, load, etc. according to the definition of the parent class. 
Completed part of the documentation on the EM algorithm of the IRT model in the DOC. 
Completed The part about the IRT model EM algorithm in test and example

#### Pull request type
- [ ] [DATASET] Add a new dataset
- [ ] [BUGFIX] Bugfix
- [ ] [FEATURE] New feature (non-breaking change which adds functionality)
- [ ] [BREAKING] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] [STYLE] Code style update (formatting, renaming)
- [ ] [REFACTOR] Refactoring (no functional changes, no api changes)
- [ ] [BUILD] Build related changes
- [ ] [DOC] Documentation content changes
- [ ] [Sync] Synchronization with a repository 
- [ ] [OTHER] Other (please describe): 

#### Changes
- Feature1, tests, (and when applicable, API doc)
- Feature2, tests, (and when applicable, API doc)

or

- Fix1, tests
- Fix2, tests

### Does this close any currently open issues?
...

### Any relevant logs, error output, etc?
...

## Checklist ##
Before you submit a pull request, please make sure you have to following:

### Essentials ###
- [ ] PR's title starts with a category (e.g. [BUGFIX], [FEATURE], [BREAKING], [DOC], etc)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage and al tests passing
- [ ] Code is well-documented (extended the README / documentation, if necessary)
- [ ] If this PR is your first one, add your name and github account to [AUTHORS.md](../blob/master/AUTHORS.md)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by SiHhangChen** [comment]
thanks

File: EduCDM/IRT/__init__.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# coding: utf-8
+# 2021/4/23 @ tongshiwei
+
+
+from .GD import IRT as GDIRT
```
Comment (LegionKing) [inline EduCDM/IRT/__init__.py:5]: 基于GD的训练尚未完成，先把相关代码注释

File: EduCDM/IRT/EM/IRT.py [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+            a, b, c = update_irt(a, b, c, self.D, self.prof,  R, r_ek, s_ek, lr, epoch_m, epsilon)
+            change = max(np.max(np.abs(a - a_tmp)), np.max(np.abs(b - b_tmp)), np.max(np.abs(c - c_tmp)),
+                         np.max(np.abs(prior_dis_tmp - prior_dis_tmp)))
+            if iteration > 20 and change < epsilon:
+                break
+        self.a, self.b, self.c, self.prior_dis = a, b, c, prior_dis
+        self.stu_prof = self.Get_stu_ability(R)
+
+
+    def predict_proba(self) -> pd.DataFrame:
```
Comment (LegionKing) [inline EduCDM/IRT/EM/IRT.py:206]: 缺少val_data，需要预测的是val_data里的答题结果

File: EduCDM/IRT/EM/IRT.py [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+        Args:
+            None
+        Return value:
+            pred_score: the probability of students response correctly
+        """
+        pred_score = irt3pl(np.sum(self.a * (np.expand_dims(self.stu_prof, axis=1) - self.b), axis=-1), 1, 0, self.c)
+        return pd.DataFrame(pred_score)
+        
+        
+    def pridict(self) -> pd.DataFrame:
```
Comment (LegionKing) [inline EduCDM/IRT/EM/IRT.py:219]: 缺少val_data，需要预测的是val_data里的答题结果

**Comment by SiHhangChen** [comment]
pytest had passed, predict and predict_proba function added parameter
 'val_data '

**Pull request merged**